app.js

```diff
@@ -559,14 +572,25 @@ Promise.all([findCompilers(), aws.initConfig(awsProps)])
 
         logger.info("=======================================");
         logger.info(`Listening on http://${hostname || 'localhost'}:${port}/`);
-        logger.info(`  serving static files from '${staticDir}'`);
         if (gitReleaseName) logger.info(`  git release ${gitReleaseName}`);
 
         function renderConfig(extra) {
             const options = _.extend(extra, clientOptionsHandler.get());
             options.compilerExplorerOptions = JSON.stringify(options);
-            options.root = versionedRootPrefix;
             options.extraBodyClass = extraBodyClass;
+            options.require = function (path) {
+                if (isDevMode()) {
+                    //this will break assets in dev mode for now
+                    return '/dist/' + path;
+                }
+                if(staticManifest.hasOwnProperty(path)) {
+                    return "dist/" + staticManifest[path];
+                }
+                if(assetManifest.hasOwnProperty(path)) {
+                    return "dist/assets/" + assetManifest[path];
+                }
+                logger.warn("Requested an asset I don't know about");
```

Comment:
Nitpick: add the path here, else if we ever see it in the warnings we won't know what the missing path was :)